src/contexts/CoursesContext.jsx

```diff
@@ -0,0 +1,25 @@
+// ============================================================================================
+// IMPORTS
+// ============================================================================================
+
+import { createContext, useState } from "react";
+import PropTypes from "prop-types";
+import { UserContextProvider } from "./UserContext";
+
+const CoursesContext = createContext(null);
+
+function CoursesContextProvider({ children }) {
+  const [coursesInfo, setCoursesInfo] = useState(null);
+
+  return (
+    <CoursesContext.Provider value={{ coursesInfo, setCoursesInfo }}>
+      {children}
+    </CoursesContext.Provider>
+  );
+}
+
+UserContextProvider.propTypes = {
+  children: PropTypes.object
```

Comment:
Instead of using PropTypes.object, it’s better to use PropTypes.node, which is more accurate for React children since it can include everything that can be rendered, not just objects.

Reply:
I agree it can add more flexibility